src/Service/DashboardHelper.php

```diff
@@ -0,0 +1,197 @@
+    {
+        $gridInformation = [];
+
+        $gridInformation = $this->addUserColumnCaseInformation($user, $gridInformation);
+
+        $gridInformation = $this->addBoardsColumnCaseInformation($municipality, $gridInformation);
+
+        return $gridInformation;
+    }
+
+    private function addUserColumnCaseInformation(User $user, array $gridInformation): array
+    {
+        // Detect which filter option current user is
+        $userFilterOption = $this->getCurrentUserFilterOption($user);
+
+        // Construct the filter urls and do the counts
+        // In hearing
+        $hearingUrl = $this->router->generate('case_index', ['case_filter' => [
+            'assignedTo' => $userFilterOption,
+            'specialStateFilter' => CaseSpecialFilterStatuses::IN_HEARING,
+        ]]);
+
+        // TODO: Update beneath when hearing stuff has been implemented
+        $hearingCount = $this->caseRepository->count(['assignedTo' => $user]);
+
+        $hearingData = [
+            'url' => $hearingUrl,
+            'count' => $hearingCount,
+        ];
+
+        // Has new party submission
+        $newPartySubmissionUrl = $this->router->generate('case_index', ['case_filter' => [
+            'assignedTo' => $userFilterOption,
+            'specialStateFilter' => CaseSpecialFilterStatuses::NEW_HEARING_POST,
+        ]]);
+
+        // TODO: Update beneath when hearing stuff has been implemented
+        $newPartySubmissionCount = $this->caseRepository->count(['assignedTo' => $user]);
+
+        $newPartySubmissionData = [
+            'url' => $newPartySubmissionUrl,
+            'count' => $newPartySubmissionCount,
+        ];
+
+        // On agenda
+        $agendaUrl = $this->router->generate('case_index', ['case_filter' => [
+            'assignedTo' => $userFilterOption,
+            'specialStateFilter' => CaseSpecialFilterStatuses::ON_AGENDA,
+        ]]);
+
+        $agendaCount = $this->caseRepository->findCountOfCasesWithUserAndWithActiveAgenda($user);
+
+        $agendaData = [
+            'url' => $agendaUrl,
+            'count' => $agendaCount,
+        ];
+
+        // Has exceeded one or more deadlines
+        $exceededDeadlineUrl = $this->router->generate('case_index', ['case_filter' => [
+            'assignedTo' => $userFilterOption,
+            'deadlines' => CaseDeadlineStatuses::SOME_DEADLINE_EXCEEDED,
+        ]]);
+
+        $exceededDeadlineCount = $this->caseRepository->findCountOfCasesWithUserAndSomeExceededDeadline($user);
+
+        $exceededDeadlineData = [
+            'url' => $exceededDeadlineUrl,
+            'count' => $exceededDeadlineCount,
+        ];
+
+        array_push($gridInformation, [
+            'grid_label' => $this->translator->trans('My cases', [], 'dashboard'),
+            'hearing_data' => $hearingData,
+            'new_party_submission_data' => $newPartySubmissionData,
+            'agenda_data' => $agendaData,
+            'exceeded_deadline_data' => $exceededDeadlineData,
```

Comment:
Rather than having four hardcoded keys here, I think it will be better to build a generic grid which is actually a list of columns with a label and a list of rows, e.g.

```php
[
 [
  'label' => trans('My cases'),
  'count' => 87,
  'rows' => [
   [
    'label' => trans('Hearing started'),
    'url' => …,
    'count' => 42,
   ],
   …,
  ],
 ],
]
```

This will make it more dynamic and probably easier to maintain. And the template can be simplified as well.

Reply:
Updated grid as proposed.